src/layer/websocket/ws_connection.cpp

```diff
@@ -36,31 +37,42 @@ namespace libp2p::connection {
 
     if (config_->ping_interval != std::chrono::milliseconds::zero()) {
       // Set pong handler
-      ws_read_writer_->setPongHandler([wp = weak_from_this()](auto payload) {
-        if (auto self = wp.lock()) {
-          self->onPong(payload);
-        }
-      });
+      using boost::beast::websocket::frame_type;
+      ws_.control_callback(
+          [weak = weak_from_this()](frame_type type,
+                                    boost::beast::string_view payload) {
+            if (type != frame_type::pong) {
+              return;
+            }
+            if (auto self = weak.lock()) {
+              self->onPong(bytestr(payload));
+            }
+          });
 
       ping_handle_ = scheduler_->scheduleWithHandle(
           [wp = weak_from_this()]() mutable {
             if (auto self = wp.lock(); self and self->started_) {
               ++self->ping_counter_;
 
-              auto payload = gsl::make_span(
-                  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-                  reinterpret_cast<const uint8_t *>(&self->ping_counter_),
-                  sizeof(self->ping_counter_));
               // Send ping
-              self->ws_read_writer_->ping(payload);
+              self->ws_.async_ping(
+                  {
+                      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                      reinterpret_cast<const char *>(&self->ping_counter_),
+                      sizeof(self->ping_counter_),
+                  },
+                  [](boost::system::error_code) {});
 
               // Start timer of pong waiting
               self->ping_timeout_handle_ =
                   self->scheduler_->scheduleWithHandle([wp] {
                     if (auto self = wp.lock()) {
-                      self->ws_read_writer_->close(
-                          websocket::WsReadWriter::ReasonOfClose::
-                              PINGING_TIMEOUT);
+                      self->ws_.async_close(
+                          {
```

Comment:
Please, apply clang-format

Reply:
It's result from both `clang-format-12` and `clang-format-15`